chain/dependencies.go

```diff
@@ -287,3 +287,6 @@ type AuthFactory interface {
 	MaxUnits() (bandwidth uint64, compute uint64)
 	Address() codec.Address
 }
+
+// RegistryFactory is the factory function, provided to the VM initializer that provides the registries for teh actions, auth and output.
+type RegistryFactory func() (actionRegistry ActionRegistry, authRegistry AuthRegistry, outputRegistry OutputRegistry)
```

Comment:
Would it be better to have a struct here? 
```go
type Registry struct {
    action ActionRegistry
    auth   AuthRegistry
    output OutputRegistry
}
```
```suggestion
type RegistryFactory func() Registry
```
I'm not sure if we plan on extending this, but I think backwards-compatible changes would be easier with a struct.

Reply:
sounds good to me.

Reply:
done